lib/WebService/Async/CustomerIO.pm

```diff
@@ -0,0 +1,320 @@
+package WebService::Async::CustomerIO;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::Async::CustomerIO - unofficial support for the Customer.io service
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+=cut
+
+use parent qw(IO::Async::Notifier);
+
+use mro;
+use Syntax::Keyword::Try;
+use Future;
+use Net::Async::HTTP;
+use Carp qw();
+use JSON::MaybeUTF8 qw(:v1);
+
+use WebService::Async::CustomerIO::Customer;
+use WebService::Async::CustomerIO::RateLimiter;
+use WebService::Async::CustomerIO::Trigger;
+
+use constant {
+    TRACKING_END_POINT                => 'https://track.customer.io/api/v1',
+    API_END_POINT                     => 'https://api.customer.io/v1/api',
+    REQUEST_PER_SECOND_LIMIT_TRACKING => 30,
+    REQUEST_PER_SECOND_LIMIT_API      => 10,
+};
+
+=head2 new
+
+Creates a new api client object
+
+Usage: C<< new(%params) -> obj >>
+
+Parameters:
+
+=over 4
+
+=item * C<site_id>
+
+=item * C<api_key>
+
+=back
+
+=cut
+
+sub configure {
+    my ($self, %args) = @_;
+    for my $k (qw(site_id api_key)) {
+        Carp::croak "Missing required argument: $k" unless exists $args{$k};
```

Comment:
This isn't going to work - any time someone calls `->configure` this will fail if they don't repeat the `site_id` and `api_key` values.

Reply:
You mean in case if somebody in sub class decided to redefine attribute? and I need to check value in `$self` as well.

Reply:
`Adjust the named parameters of the Notifier as given by the %params hash.`. Got it, I misunderstood the idea of this method, I thought it's only called by constructor.